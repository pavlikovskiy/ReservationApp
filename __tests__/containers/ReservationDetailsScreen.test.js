import React from "react";
import wait from "waait";
import { MockedProvider } from "react-apollo/test-utils";
import renderer from "react-test-renderer";

import {
  ReservationDetails,
  GET_RESERVATION_QUERY
} from "../../app/containers/ReservationDetailsScreen";

const mocks = [
  {
    request: {
      query: GET_RESERVATION_QUERY,
      variables: {
        id: "cjuatrgg0eggt0b03az2hakfx"
      }
    },
    result: {
      data: {
        reservation: {
          name: "Jeff McCall",
          arrivalDate: "9/9/2019",
          hotelName: "Hilton",
          id: "cjuatrgg0eggt0b03az2hakfx",
          departureDate: "9/9/2020"
        }
      }
    }
  }
];

describe("ReservationDetailsScreen", () => {
  it("should render loading message ", () => {
    const component = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ReservationDetails id="cjuatrgg0eggt0b03az2hakfx" />
      </MockedProvider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should render with data from graph mock", async () => {
    const component = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ReservationDetails id="cjuatrgg0eggt0b03az2hakfx" />
      </MockedProvider>
    );

    await wait(1); // wait to resolve promise
    expect(component.toJSON()).toMatchSnapshot();
  });
});
