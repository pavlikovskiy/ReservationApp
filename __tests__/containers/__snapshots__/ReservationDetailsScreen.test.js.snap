// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationDetailsScreen should render loading message  1`] = `
<Text>
  Reservation details is loading...
</Text>
`;

exports[`ReservationDetailsScreen should render with data from graph mock 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "#FAFAFA",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingBottom": 20,
      "paddingLeft": 20,
    }
  }
>
  <Text
    style={
      Object {
        "alignItems": "center",
        "color": "#808080",
        "flex": 1,
        "fontSize": 14,
        "fontWeight": "bold",
        "justifyContent": "center",
        "paddingBottom": 20,
        "paddingHorizontal": 10,
        "paddingTop": 10,
        "paddingVertical": 10,
      }
    }
  >
    Jeff McCall
     at 
    Hilton
  </Text>
  <Text
    style={Object {}}
  >
    Name: 
    Jeff McCall
  </Text>
  <Text
    style={Object {}}
  >
    Hotel Name: 
    Hilton
  </Text>
  <Text
    style={Object {}}
  >
    From: 
    9/9/2019
  </Text>
  <Text
    style={Object {}}
  >
    To: 
    9/9/2020
  </Text>
</View>
`;
